projects/plugins/jetpack/extensions/blocks/paywall/edit.js

```diff
@@ -46,18 +55,82 @@ function PaywallEdit( { className } ) {
 			<InspectorControls>
 				<PanelBody
 					className="jetpack-subscribe-newsletters-panel"
-					title={ __( 'Newsletter visibility', 'jetpack' ) }
+					title={ __( 'Content visibility', 'jetpack' ) }
 					icon={ <JetpackEditorPanelLogo /> }
 					initialOpen={ true }
 				>
-					<NewsletterAccessDocumentSettings
-						accessLevel={ accessLevel }
-						setPostMeta={ setPostMeta }
-					/>
+					<p>
+						{ __(
+							'Choose who will be able to read the content below the paywall block.',
+							'jetpack'
+						) }
+					</p>
+					<PaywallBlockSettings accessLevel={ accessLevel } setPostMeta={ setPostMeta } />
 				</PanelBody>
 			</InspectorControls>
 		</>
 	);
 }
 
+function PaywallBlockSettings( { accessLevel, setPostMeta } ) {
+	const { hasNewsletterPlans, stripeConnectUrl, isLoading } = useSelect( select => {
+		const { getNewsletterProducts, getConnectUrl, isApiStateLoading } = select(
+			'jetpack/membership-products'
+		);
+		return {
+			isLoading: isApiStateLoading(),
+			stripeConnectUrl: getConnectUrl(),
+			hasNewsletterPlans: getNewsletterProducts()?.length !== 0,
+		};
+	} );
+
+	const postVisibility = useSelect( select => select( editorStore ).getEditedPostVisibility() );
+
+	if ( isLoading ) {
+		return (
+			<Flex direction="column" align="center">
+				<Spinner />
+			</Flex>
+		);
+	}
+
+	let _accessLevel = accessLevel ?? accessOptions.subscribers.key;
+	if ( _accessLevel === accessOptions.everybody.key ) {
+		_accessLevel = accessOptions.subscribers.key;
+	}
+
+	const accessLabel = accessOptions[ _accessLevel ]?.label;
+
+	const showMisconfigurationWarning = getShowMisconfigurationWarning( postVisibility, accessLevel );
+
+	return (
+		<PostVisibilityCheck
+			render={ ( { canEdit } ) => (
+				<PanelRow className="edit-post-post-visibility">
+					<Flex direction="column">
+						{ showMisconfigurationWarning && <MisconfigurationWarning /> }
+						<FlexBlock direction="row" justify="flex-start">
+							{ canEdit && (
+								<div className="editor-post-visibility">
+									<NewsletterAccessRadioButtons
+										isEditorPanel={ true }
+										onChange={ setPostMeta }
+										accessLevel={ _accessLevel }
+										stripeConnectUrl={ stripeConnectUrl }
+										hasNewsletterPlans={ hasNewsletterPlans }
+										paywallBlockSettings={ true }
+									/>
```

Comment:
Questions: 
   - If I'm reading this right, we're re-using the NewsletterAccessRadioButtons component within the Paywall block?
   - Does it make sense to show options for 'Everyone' and 'All Subscribers' for a Paywall block? I would think the point of the block is to put the content below behind a paywall for paid subscribers?